File: Leetcode/869 - Reordered Power of 2/solution.cpp
bool reorderedPowerOf2(int n) {
    string strNum = to_string(n);
    sort(strNum.begin(), strNum.end());

    for (int i = 1; i < 1e9; i <<= 1) {
        string stri = to_string(i);
        sort(stri.begin(), stri.end());
        if (stri == strNum) return true;
    }

    return false;
}

File: Leetcode/869 - Reordered Power of 2/solution.ts
function reorderedPowerOf2(n: number): boolean {
    const strNum: string = n.toString().split('').sort().join('');

    for (let i: number = 1; i <= 1e9; i <<= 1) {
        const stri: string = i.toString().split('').sort().join('');
        if (stri === strNum) {
            return true;
        }
    }
    return false;
};